Scripts/Runtime/Entities/Lifecycle/AbstractEntityLifecycleStatusSystem.cs

```diff
@@ -0,0 +1,303 @@
+                m_WorldCreatedEntities = new NativeList<Entity>(ChunkUtil.MaxElementsPerChunk<Entity>(), Allocator.Persistent);
+                m_WorldDestroyedEntities = new NativeList<Entity>(ChunkUtil.MaxElementsPerChunk<Entity>(), Allocator.Persistent);
+                m_WorldCleanupEntities = new NativeParallelHashSet<Entity>(ChunkUtil.MaxElementsPerChunk<Entity>(), Allocator.Persistent);
+                m_CleanupFrame = new NativeReference<int>(-1, Allocator.Persistent);
+
+                m_CleanupEntityQuery = m_World.EntityManager.CreateEntityQuery(EntityCleanupHelper.CLEAN_UP_ENTITY_COMPONENT_TYPE);
+            }
+
+            private void Dispose()
+            {
+                m_AccessController.Dispose();
+                m_WorldEntitiesState.Dispose();
+                m_WorldCreatedEntities.Dispose();
+                m_WorldDestroyedEntities.Dispose();
+                m_WorldCleanupEntities.Dispose();
+                m_CleanupFrame.Dispose();
+                m_CleanupEntityQuery.Dispose();
+            }
+
+            private void AddLifecycleStatusSystem(AbstractEntityLifecycleStatusSystem lifecycleStatusSystem)
+            {
+                m_LifecycleStatusSystems.Add(lifecycleStatusSystem);
+            }
+
+            public void RemoveLifecycleStatusSystem(AbstractEntityLifecycleStatusSystem lifecycleStatusSystem)
+            {
+                m_LifecycleStatusSystems.Remove(lifecycleStatusSystem);
+                if (m_LifecycleStatusSystems.Count == 0)
+                {
+                    Dispose();
+                }
+            }
+
+            public JobHandle AcquireCreatedAndDestroyedEntities(
+                JobHandle dependsOn,
+                out NativeArray<Entity>.ReadOnly createdEntities,
+                out NativeArray<Entity>.ReadOnly destroyedEntities,
+                bool shouldIncludeCleanupEntities)
+            {
+                m_AccessController.Acquire(AccessType.ExclusiveWrite);
+                
+                m_World.EntityManager.GetCreatedAndDestroyedEntities(m_WorldEntitiesState, m_WorldCreatedEntities, m_WorldDestroyedEntities);
+                
+                if (shouldIncludeCleanupEntities)
+                {
+                    NativeArray<Entity> cleanupEntities = m_CleanupEntityQuery.ToEntityArrayAsync(Allocator.TempJob, out JobHandle cleanupDependsOn);
+
+                    IncludeCleanupEntitiesJob includeCleanupEntitiesJob = new IncludeCleanupEntitiesJob(
+                        m_WorldCleanupEntities,
+                        m_WorldDestroyedEntities,
+                        cleanupEntities,
+                        m_CleanupFrame,
+                        UnityEngine.Time.frameCount);
+                    cleanupDependsOn = includeCleanupEntitiesJob.Schedule(cleanupDependsOn);
+                    cleanupEntities.Dispose(cleanupDependsOn);
+                    cleanupDependsOn.Complete();
```

Comment:
curious why you have to complete this. Can't you just roll the dependency into `dependsOn` and pass it back to the caller?

Reply:
oh, I see you need those `m_WorldCreatedEntities` and `m_WorldDestroyedEntities` resolved right away. Why are you passing through a `dependsOn` and treadting this like it's async then? Is anything actually touching `dependsOn` in here? Something fishy going on...

Reply:
Good catch on the fishyness, I've updated to properly combine and complete the incoming job handles.